### DISCUSSION TITLE
Insert "sent" data into database after a message is successfully sent

### DISCUSSION THREAD

**Discussion by LanciaH** [nextgenhealthcare/connect#5774] [Q&A]
Hi All,

Very much a newbie here.  I have a channel that sends data, everything is working as expected.  

The vendor has asked me to insert the data sent into an oracle database after successful sending of the data, statuscode = 200.

{
  "data" : true,
  "totalRecords" : 1,
  "statusCode" : 200,
  "error" : ""
}

Not Sure where to start.  In the channel, under the destination tab,  I created another Destination to handle the database insert.  I only want this destination to fire when there is a statuscode of 200.  Pretty sure I have to create a "filter" that return a true?    Also in the insert how do I reference the values sent?  Is there an example I can download? 

Thanks

**Comment by pacmano1** [comment]
The typical place you would do that is in a response transformer.    That does mean making a databae call with javacript in the response transformer itself.

**Reply by pacmano1** [reply]
You can of course do it in a new destiantion - and your filter idea is correct.  

You can always get a message from another destination like this (in this case the destination normally expect JSON as a response and you want to create a JS object from that response.

`var the_other_response = JSON.parse(responseMap.get('The Destination').getMessage())`

**Comment by LanciaH** [comment]
Thanks Pacman,  I have the JS built just not sure on how to reference the message sent and on how to trigger the event.
...
![f3](https://user-images.githubusercontent.com/104143486/236894698-9ee79a29-3777-468f-a043-c136d6ae9ea5.jpg)
![f2](https://user-images.githubusercontent.com/104143486/236894699-89d88dc5-c025-4acf-8ee5-e4f5663134a2.jpg)
![f1](https://user-images.githubusercontent.com/104143486/236894702-cb5ca16b-dcc8-403c-a5c2-5d159c91cd0d.jpg)

...

**Reply by pacmano1** [reply]
Edit your post, and use code blocks and format the code.  Please read:

https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks

**Comment by mjtrac** [comment]
In Mirth, the response received from any destination is stored in a response object named with the letter d and the destination's number.  You also have access to response information by adding a response transformer to the destination. 

So you can add a destination to your channel in which you do the work up updating a database based on the response you received from the prior destination.  Every one of your destinations will also have access to the message as it emerged from your source connector.  Make sure your new destination waits for the prior destination before firing (this is the default behavior.)

(This is an exercise in the fundamentals course, btw.)

**Reply by pacmano1** [reply]
@mjtrac , good point on the course - I've heard great things about it.

(I do not work for Nextgen and have heard that feedback many times)

**Comment by LanciaH** [comment]
Thanks for the suggestion about the training.  I've asked my manager about Mirth Training, we will see.

I have my database inserts working correctly referencing the data fields in the sent channel.

I am having a problem referencing the statuscode of the first destination channel.
My Filter Code

```

var returnmsg= $('d1').message;	//POST
logger.info($('d1').message)
if (returnmsg && returnmsg != ''){
	var returnjmsg = JSON.parse(returnmsg)
      if (returnjmsg['statusCode']!='200' || returnjmsg['error']!=''){
		logger.info('PatientFlags: ERROR: ' + returnmsg)
		return false;
	}
	return true;
}
return false;

```
My error is apparently on line 297, but 297 is not part of my code.

```

297: msg = JSON.parse(connectorMessage.getTransformedData());
298: function filterRule1() {
299: var returnmsg= $('d1').message;	//POST
300: logger.info($('d1').message)
301: 
LINE NUMBER:	297
DETAILS:	SyntaxError: Unexpected token: <
	at 578a4691-8cee-49cd-8dd3-d1ce68be9807_JavaScript_Filter_Transformer_6:297 (doScript)
	at 578a4691-8cee-49cd-8dd3-d1ce68be9807_JavaScript_Filter_Transformer_6:341
	at com.mirth.connect.server.transformers.JavaScriptFilterTransformer$FilterTransformerTask.doCall(JavaScriptFilterTransformer.java:235)
	at com.mirth.connect.server.transformers.JavaScriptFilterTransformer$FilterTransformerTask.doCall(JavaScriptFilterTransformer.java:187)
	at com.mirth.connect.server.util.javascript.JavaScriptTask.call(JavaScriptTask.java:114)
	at java.util.concurrent.FutureTask.run(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)

```

**Reply by jonbartels** [reply]
`JSON.parse(connectorMessage.getTransformedData());` and `SyntaxError: Unexpected token: <` 

Check your data types. Mirth is trying to interpret a message as JSON but the message content is XML.

**Reply by LanciaH** [reply]
Hi Jon,
That makes perfect sense.  Thanks!  
Is there a way to "put" the XML message into a JSON array using the XML tags?

**Reply by LanciaH** [reply]
Hi Jon,
Another suggestion was to set the data type of the response in the SEND Medication Channel.  I tried a couple of different scenarios but they all end with the same response.

All I want is the response from the previous destination channel, "Send Medication"
The error from their server ooks like

**HTTP Sender error
ERROR MESSAGE: {"data":false,"totalRecords":0,"statusCode":400,"error":"2000624446079710: VALIDATION_FAILURE: Only DcDate, DcReason and OnHold are allowed to be updated.;"}**

I have limited my transformer code to 
var resp = $('d1').getMessage();
logger.info(resp); 
and I still get the same error.

`Transformer error
ERROR MESSAGE: Error evaluating transformer
com.mirth.connect.server.MirthJavascriptTransformerException: 
CHANNEL:	Fusion_PatientMeds- New Orders-Audit 4
CONNECTOR:	Insert audit row to emar_patient_flags2
SCRIPT SOURCE:	
SOURCE CODE:	
292:         }
293:         eval('importPackage(' + Packages.java.lang.Class.forName(className).getPackage().getName() + ')');
294:     }
295: }
296: function doScript() {
297: msg = JSON.parse(connectorMessage.getTransformedData());
298: function doFilter() { phase[0] = 'filter'; return true; }function doTransform() { phase[0] = 'transformer'; logger = Packages.org.apache.log4j.Logger.getLogger(phase[0]);
299: 
300: 
301: var resp = $('d1').getMessage();
LINE NUMBER:	297
DETAILS:	SyntaxError: Unexpected token: <`

**Reply by pacmano1** [reply]
This is really much eaiser in a javascript writer.  Also, it would be far better to return the ID of the inserted row as part of your insert and add that to your audit table.

Both the insert to the table, returning the insert ID and doing the update in the same place.  

```js
var dbConn = 'you have this already';
var sql =  'insert into ... returning ID'   // this varies by SQL engine
$c('SQLInsertRow',sql) // put this in the channel map so you can see the actual sql that will be called.
var result = ... (call the DB here like you are, except you expect a result) like dbConn.executeUpdate(sql)
result.next()  // you can add protection code, but you gotta ask, how likely is it to fail, so we will assume it always works
var id = result.getInt(1) // assume the ID is an integer.

sql = 'insert into audit table (date, id) values (now(), {id}'
$c('SQLInsertAudit',sql) // put this in the channel map so you can see the actual sql that will be called.
result = .... (call the dbConn)
```

**Reply by LanciaH** [reply]
I'm sure you are right but I barely understand Mirth as it is and the database insert is the only thing working.  All I want to do is check the response of the previous channel.  Maybe once I get the response portion working I'll combine both channels into one neat channel.

**Comment by mjtrac** [comment]
Before Mirth Connect calls your transformer code it executes about 100 lines of setup in JavaScript, which is why your line number is surprising.  But the complaint of an unexpected token "<" is saying Mirth is trying to parse the data which came back in your connector's response.  By changing your anticipated response type to Raw on inbound and outbound for the destination in which the error is occurring, you'll turn off the connector's attempt to parse the response.  Or, probably, setting the data type to XML will allow the response transformer to treat the entire response as XML, from which you can extract anything embedded in the response.

You can change anticipated response types by clicking Set Data Types on your summary screen.

The error is being generated on your "Insert audit row..." destination so that is the one where you'd want to set your data type to XML (or raw).   On line 297 of the Javascript, "msg = JSON.parse(connectorMessage.getTransformedData());" is what triggers the error, and unless you have written such a line, that is Mirth Connect attempting to treat your response data as JSON, when it is coming back to you as XML.  You can also change how your HTTP Sender handles the response, but this may cause you additional problems.  There is a choice of "Plain Body" or "XML Body" for response content on the HTTP Sender.

Most likely, you are getting back an XML response body, which has JSON embedded inside an XML node, but are passing the root node of your XML into JSON.parse().

**Reply by LanciaH** [reply]
I moved checking the response to the "send medication " channel,  figuring my insert was not the place to check the response of the previous channel.

Is the code below not valid?
**var resp = JSON.parse($('d1').message);
logger.info(resp);**

I have gotten past the < syntax error 
`Response Transformer error
ERROR MESSAGE: Error evaluating response transformer
com.mirth.connect.server.MirthJavascriptTransformerException: 
CHANNEL:	Fusion_PatientMeds- New Orders-Audit 4
CONNECTOR:	Send Medication
SCRIPT SOURCE:	response
SOURCE CODE:	
297: msg = JSON.parse(connectorMessage.getResponseTransformedData());
298: function doTransform() {
299: 
300: 
301: var obj={}; //create object
302: var resp = JSON.parse($('d1').message);
303: logger.info(resp); // print it
304: 
305: //var obj={}; //create object
306: //obj. name=$('d1').message;
LINE NUMBER:	302
DETAILS:	SyntaxError: Unexpected token: u`

**Comment by mjtrac** [comment]
Perhaps try this or similar, and tell us what gets logged: 

var destination1Response = responseMap.get('d1');
logger.info("Destination 1 response info: status message, error, message");
logger.info( destination1Response.getStatusMessage() ) ;
logger.info( destination1Response.getError() );
logger.info( destination1Response.getMessage() );
logger.info("END Destination 1 response info");

**Reply by LanciaH** [reply]
Thanks for your continued help.  The Logger at least has informations now.
`Response Transformer error
ERROR MESSAGE: Error evaluating response transformer
com.mirth.connect.server.MirthJavascriptTransformerException: 
CHANNEL:	Fusion_PatientMeds- New Orders-Audit 4
CONNECTOR:	Send Medication
SCRIPT SOURCE:	response
SOURCE CODE:	
298: function doTransform() {
299: 
300: 
301: var destination1Response = responseMap.get('Entity_Medications');
302: logger.info("Destination 1 response info: status message, error, message");
303: logger.info( destination1Response.getStatusMessage() ) ;
304: logger.info( destination1Response.getError() );
305: logger.info( destination1Response.getMessage() );
306: logger.info("END Destination 1 response info");
307: 
LINE NUMBER:	303
DETAILS:	TypeError: Cannot call method "getStatusMessage" of null
	at 0ed5ade8-2cdc-4a1c-b5e2-22cb78817e54_Response_Transformer_1:303 (doTransform)
	at 0ed5ade8-2cdc-4a1c-b5e2-22cb78817e54_Response_Transformer_1:351 (doScript)
	at 0ed5ade8-2cdc-4a1c-b5e2-22cb78817e54_Response_Transformer_1:353
	at com.mirth.connect.server.transformers.JavaScriptResponseTransformer$ResponseTransformerTask.doCall(JavaScriptResponseTransformer.java:231)
	at com.mirth.connect.server.transformers.JavaScriptResponseTransformer$ResponseTransformerTask.doCall(JavaScriptResponseTransformer.java:169)
	at com.mirth.connect.server.util.javascript.JavaScriptTask.call(JavaScriptTask.java:114)
	at java.util.concurrent.FutureTask.run(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)
`
**Logger Info**
`[2023-05-25 11:55:46.510]  ERROR  (com.mirth.connect.connectors.http.HttpDispatcher:976): Error executing response transformer for channel Fusion_PatientMeds- New Orders-Audit 4 (0ed5ade8-2cdc-4a1c-b5e2-22cb78817e54) on destination Send Medication.
com.mirth.connect.donkey.server.channel.components.ResponseTransformerException: CHANNEL:
	Fusion_PatientMeds- New Orders-Audit 4CONNECTOR:
	Send MedicationSCRIPT SOURCE:
	responseSOURCE CODE:
	298: function doTransform() {299: 300: 301: var destination1Response = responseMap.get('Entity_Medications');302: logger.info("Destination 1 response info: status message, error, message");303: logger.info( destination1Response.getStatusMessage() ) ;304: logger.info( destination1Response.getError() );305: logger.info( destination1Response.getMessage() );306: logger.info("END Destination 1 response info");307: LINE NUMBER:
	303DETAILS:
	TypeError: Cannot call method "getStatusMessage" of null
	at 0ed5ade8-2cdc-4a1c-b5e2-22cb78817e54_Response_Transformer_1:303 (doTransform)
	at 0ed5ade8-2cdc-4a1c-b5e2-22cb78817e54_Response_Transformer_1:351 (doScript)
	at 0ed5ade8-2cdc-4a1c-b5e2-22cb78817e54_Response_Transformer_1:353
	at com.mirth.connect.server.transformers.JavaScriptResponseTransformer$ResponseTransformerTask.doCall(JavaScriptResponseTransformer.java:238)
	at com.mirth.connect.server.transformers.JavaScriptResponseTransformer$ResponseTransformerTask.doCall(JavaScriptResponseTransformer.java:169)
	at com.mirth.connect.server.util.javascript.JavaScriptTask.call(JavaScriptTask.java:114)
	at java.util.concurrent.FutureTask.run(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)Caused by: com.mirth.connect.server.MirthJavascriptTransformerException: CHANNEL:
	Fusion_PatientMeds- New Orders-Audit 4CONNECTOR:
	Send MedicationSCRIPT SOURCE:
	responseSOURCE CODE:
	298: function doTransform() {299: 300: 301: var destination1Response = responseMap.get('Entity_Medications');302: logger.info("Destination 1 response info: status message, error, message");303: logger.info( destination1Response.getStatusMessage() ) ;304: logger.info( destination1Response.getError() );305: logger.info( destination1Response.getMessage() );306: logger.info("END Destination 1 response info");307: LINE NUMBER:
	303DETAILS:
	TypeError: Cannot call method "getStatusMessage" of null
	at 0ed5ade8-2cdc-4a1c-b5e2-22cb78817e54_Response_Transformer_1:303 (doTransform)
	at 0ed5ade8-2cdc-4a1c-b5e2-22cb78817e54_Response_Transformer_1:351 (doScript)
	at 0ed5ade8-2cdc-4a1c-b5e2-22cb78817e54_Response_Transformer_1:353
	at com.mirth.connect.server.transformers.JavaScriptResponseTransformer$ResponseTransformerTask.doCall(JavaScriptResponseTransformer.java:231)
	... 6 more`

**Reply by mjtrac** [reply]
The line 
> 303DETAILS: TypeError: Cannot call method "getStatusMessage" of null

suggests that lanciah is not getting back the response information using the connector name, for whatever reason, so I'd still suggest trying the letter d plus the destination number.

**Comment by mjtrac** [comment]
I think but am not sure that you don't want to use the connector's name in the responseMap.get call.  You want to look up the destination number of your connector Entity_Medications on the Destinations tab of your channel.  If it is destination 4, use responseMap.get('d4').

**Reply by pacmano1** [reply]
If you want the message from a destination:
```js
responseMap.get('destination name').getMessage()  // using the actual name of the destination here.
```

**Reply by mjtrac** [reply]
I've used d1, etc..., but have never used 'destination name').  I know that the former works, the latter perhaps works as well.

**Reply by pacmano1** [reply]
Was not correcting you, just adding an option since names for me are easier than the destination number.

**Comment by LanciaH** [comment]
I made the following change.
var destination1Response = responseMap.get**($('d1'))**;
logger.info("Destination 1 response info: status message, error, message");
logger.info( destination1Response.getStatusMessage() ) ;
logger.info( destination1Response.getError() );
logger.info( destination1Response.getMessage() );
logger.info("END Destination 1 response info");

`Response Transformer error
ERROR MESSAGE: Error evaluating response transformer
com.mirth.connect.server.MirthJavascriptTransformerException: 
CHANNEL:	Fusion_PatientMeds- New Orders-Audit 4
CONNECTOR:	Send Medication
SCRIPT SOURCE:	response
SOURCE CODE:	
298: function doTransform() {
299: 
300: 
301: var destination1Response = responseMap.get($('d1'));
302: logger.info("Destination 1 response info: status message, error, message");
303: logger.info( destination1Response.getStatusMessage() ) ;
304: logger.info( destination1Response.getError() );
305: logger.info( destination1Response.getMessage() );
306: logger.info("END Destination 1 response info");
307: 
LINE NUMBER:	303
DETAILS:	TypeError: Cannot call method "getStatusMessage" of null
	at 0ed5ade8-2cdc-4a1c-b5e2-22cb78817e54_Response_Transformer_1:303 (doTransform)
	at 0ed5ade8-2cdc-4a1c-b5e2-22cb78817e54_Response_Transformer_1:351 (doScript)
	at 0ed5ade8-2cdc-4a1c-b5e2-22cb78817e54_Response_Transformer_1:353
	at com.mirth.connect.server.transformers.JavaScriptResponseTransformer$ResponseTransformerTask.doCall(JavaScriptResponseTransformer.java:231)
	at com.mirth.connect.server.transformers.JavaScriptResponseTransformer$ResponseTransformerTask.doCall(JavaScriptResponseTransformer.java:169)
	at com.mirth.connect.server.util.javascript.JavaScriptTask.call(JavaScriptTask.java:114)
	at java.util.concurrent.FutureTask.run(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)`

**Reply by pacmano1** [reply]
```js
$c('other_destination response status', $('d1'))
$c('other_destination response value', $('d1').getMessage())
```


![image](https://github.com/nextgenhealthcare/connect/assets/44065187/2268c96b-ef47-4c9b-991d-08cc0013bf85)

**Reply by LanciaH** [reply]
by 'other_destination'  do you mean my other channel?  Which is called 'Insert audit row to emar_patient_flags2'

So 
$c('Insert audit row to emar_patient_flags2 response  status', $('d1'))
$c('Insert audit row to emar_patient_flags2 response value', $('d1').getMessage())

**Comment by mjtrac** [comment]
FWIW, there are going to be different sets of issues potentially involved here.  

You have multiple destinations on your channel.  Let's say you have two destinations which operate sequentially; there's a checkbox that is normally checked that says a destination will wait for the previous destination to complete.  If that is checked, so that destination 2 runs after destination 1 completes, you should not have a null return value when calling responseMap.get('d1') from your second destination.  So maybe your second destination doesn't have that box checked?

In addition to or instead of trying to retrieve destination 1's response object from within destination 2, you can also retrieve destination 1 response information in a response transformer placed on destination 1.  What Mirth will do is look at the inbound response data type for the response, and, based on that type, convert it as necessary.  If you get back XML, it will still be XML; if it were a pipe-hat style HL7 message, it will be converted to XML.  If JSON, it may stay JSON, I'm not certain.  But if it tries to parse XML as JSON, or vice versa you'll get an error when it attempts to do the parsing.  What pacmano1 is showing above is a response, probably to a Mirth API call, which has come back as JSON.  The JSON is just one part of the response object, the "message."

**Reply by LanciaH** [reply]
" second destination doesn't have that box checked?" It does
"a response transformer placed on destination 1" That is exactly where it is, in Destination one "Edit Response"

In my first destination channel I have the following code in the edit response and the logger works and displays what I am sending to the vendor.  All I want is the response and to be able to reference it. 

`$('Entity_Medications')

if($('Entity_Medications')!=""){
	var rsp = new XML($('Entity_Medications'));

	var success = false;
	var errorString = false;
	try{
	errorString = rsp.toString().split("<a:Errors>")[1].split("</a:Errors>")[0];
	error = true
	}
	catch(Exception){ 
		error =false;
		errorString = false;
	}

	logger.info(rsp)`

**Comment by mjtrac** [comment]
I'm not sure if you're saying your problem is solved.

I'm not sure why you feel you must create an XML object out of
$('Entity_Medications').

But if you can log out rsp or extract error information from it, the only
other thing to do to make it available to any subsequent destination is to
put it into the channelMap with channelMap.put('rsp',rsp);

If you put rsp into the channelMap, you are placing an XML object into the
channel map.  If you put rsp.toString() into the channel map you are
putting a string representation of rsp into the channel map.

If you're not sure about what type $('Entity_Medications') is returning,
you can write:
 logger.info(typeof $('Entity_Medications'))

**Reply by LanciaH** [reply]
Thanks,

I really don't need the XML I was just trying anything and was hoping the response status was in there

**Comment by LanciaH** [comment]
Hi Mitch the type returned is **object.**

Let me start over.  I send the vendor medications and the vendor sends back a statusCode, 200 good, 400 bad.

I need to capture this statusCode.  If 200 do the Audit table Insert.  If 400 do nothing.

See the screen hsots

![statusCode200](https://github.com/nextgenhealthcare/connect/assets/104143486/a9915dd9-c116-4aeb-bd0a-fb5d6ab08904)
![statusCode400](https://github.com/nextgenhealthcare/connect/assets/104143486/baf32fea-e006-4d94-b0bd-1ccb9bd5aeff)

**Comment by mjtrac** [comment]
That's very helpful, thanks.

So you are getting back JSON.  You want to set the Data Type of your Send
Medication Response to JSON inbound and could also set it to JSON
outbound.  You do that off the dialog screen which appears when, on your
channel's Summary tab, you press "Set Data Types" just under the channel's
name in the "Channel Properties" frame.

Then, when you add a response transformer to your Send Medication
destination, the statusCode from the JSON will be available in the response
transformer as msg['statusCode'].

In the response transformer, to put the value into the channel map for use
by later destinations:
    channelMap.put('statusCode', msg['statusCode']); // or
msg['statusCode']*.toString()
*if needed

In any later destination to get the statusCode back,
channelMap.get('statusCode');

**Comment by mjtrac** [comment]
(You add your response transformer to the Send Medications destination by first highlighting that destination in the destinations tab, then selecting Edit Response from the Channel Tasks menu to the left, then clicking Add New Step in the Transformer Tasks menu and either putting msg['statusCode'] in as your mapping and statusCode as your variable name, or changing the step from Mapper (what you get when you click Add New Step) to JavaScript and entering JS as in the comment above.)

**Reply by pacmano1** [reply]
I am gonna beat a dead horse, the response, the response status, etc.  from other destinations is always avaiable without having to put it in a variable.

**Reply by LanciaH** [reply]
Okay, I believe you, How do I reference it?

**Reply by mjtrac** [reply]
While pacmano1 is correct, I think you might find the response transformer easier.  Is it possible that in error responses the response object's getError method would be needed, while in OK responses, the getMessage method would be needed instead, and might the appropriate JSON be automatically placed into msg, by Mirth, for the response transformer?

**Comment by LanciaH** [comment]
Mitch, This is perfect, This is all I needed in the first place.  Thanks so much for our help. 
In the "Send Medications" Channel Edit Response I have the following
```
channelMap.put('vStatusCode', msg['statusCode']);
logger.info(msg['statusCode']);
```

In the next Channel, **Insert audit row to emar_patient_flags2**, I reference vStatusCode 
```var dbConn;
var result;
var newInstructions;

if ($('vStatusCode')==200){

try {
	newInstructions = $('vinstructions').replace("'","''");
	newInstructions = newInstructions.replace("&gt",">");
	dbConn = DatabaseConnectionFactory.createDatabaseConnection('oracle.jdbc.driver.OracleDriver','jdbc:oracle:thin:@(DESCRIPTION=(ADDRESS=(PROTOCOL=TCP)(HOST=*****************)(PORT=1521))(CONNECT_DATA=(SERVER=DEDICATED)(SERVICE_NAME=**************.us)(FAILOVER_MODE=(TYPE=select)(METHOD=basic))))','****',*****');

	result = dbConn.executeUpdate(" Insert into OITDELA.FUSION_AUDIT_MEDS (STARTDATE, STOPDATE, AUTHPROVIDER, ROUTE, EXTERNALID, INSTRUCTIONS, DOSE, DOSEUNIT, EXTERNALPATIENTID,  EXTERNALMEDICATIONID, COMMENTS, CREATE_DATE)  values ('"+
	   $('vstartdate')+"','"+$('vstopdate')+"','"+$('vauthprovider')+"','"+$('vroute')+"','"+$('vexternalid')+"','"+newInstructions+"','"+$('vdose')+"','"+$('vdoseunit')+"','"+$('vexternalpatientid')+"','"+$('vexternalmedicationid')+"','"+$('vcomments').replace("'","''")+"',sysdate)");
	

} finally {
	if (dbConn) { 
		dbConn.close();
	}
}
} //END STATUSCODE =200 IF```

**Reply by pacmano1** [reply]
Those are not "channels" they are "destinations" of a single channel.

**Comment by LanciaH** [comment]
Now getting back to pacmano1's suggestion.  Can I put my database insert under the first Channel's edit response?  And get rid of the second channel?

**Comment by mjtrac** [comment]
Having the additional destination would, among other things, give you an
easy way to check counts, spot failures, etc.... But you can put JavaScript
in the Response transformer if you wish, and have it do whatever you want.
Think of Mirth as a framework where you can use services it provides while
rolling your own custom needs in JavaScript.

**Reply by LanciaH** [reply]
Okay thanks, I'll leave it as is for now.  Thanks for everyones help.
Ralph
